aip/auth/4115.md

```diff
@@ -5,124 +5,110 @@ state: approved
 
-```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=comma-separated-list-of-scopes
-```
-
-The auth library **should** allow the caller to optionally specify a list of custom scopes,
-and add the `scopes` parameter to the request when needed.
-Depending on the runtime environment, the request for custom scopes will be transparently
-ignored by the server (GCE) or fulfilled (GCE equivalent).
+1. Check if the remote endpoint supports mTLS. 
+   * If yes, go to step (2).
+   * If not, go to step (3). 
+2. Send a request to the MDS mTLS configuration endpoint. If the request is successful and the 
+   response contains a Secure Session Agent address, use the address to access Google default mTLS
+   credentials, and go to step (4). If the request fails or the response contains an empty address,
+   go to step (3).
+3. Fall back to TLS [END].
+4. Configure the TLS library to use the Secure Session Agent ([example][3]) for client authentication
+   during the mTLS handshake.
 
-### Google App Engine Standard 1.0 Runtime
+### Request Bound Tokens
 
-If the application runs on a App Engine Standard 1st generation instance, the
-auth library **should** rely on the [Google App Engine SDK][3] to retrieve the
-access token. Here is [one example][5] in Go. Essentially the SDK relies on the
-underlying App Engine Identity service to generate the token.
+To access Google APIs with bound tokens, the client **should** request tokens from MDS. All virtual 
+environments MDS expose a HTTP endpoint (access token endpoint) that serves access tokens.
 
-Like GCE equivalent runtimes, the auth library can specify scopes when requesting the
-token in the App Engine Standard 1.0 runtime.
+The MDS access token endpoint is: 
+```
+http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token
+```
 
-Sample code of getting the access token in Go:
+The request to the MDS endpoint **should** be a HTTP GET request. The request **may** have a “scopes” 
+URL parameter with a list of comma-separated scopes. The auth library **should** allow the caller to 
+optionally specify a list of custom scopes, and add the “scopes” parameter to the request when needed. 
+Depending on the runtime environment, the request for custom scopes will be transparently ignored by 
+the server or fulfilled.
 
-```go
-import "google.golang.org/appengine"
-...
-token, exp, err := appengine.AccessToken(context, scopes...)
-...
-```
+The response from the MDS endpoint **should** contain an access token.
 
-### Token Expiration
+The client **must** follow the steps below to request new access tokens for Google APIs if existing 
+tokens expire. Old valid tokens can be reused directly. 
 
-The access tokens expire after a short period of time. The auth library **must**
-get a new token if the existing token expires.
+1. Send a HTTP request to the MDS access token endpoint, and use the token and go to step (2).
```

Comment:
Suggested edit: s/and use the token/retrieve the access token from the response

Reply:
Done.